ladybug/dt.py

```diff
@@ -220,3 +224,199 @@ def ToString(self):
     def __repr__(self):
         """Return date time as a string."""
         return self.__str__()
+
+
+class Date(date):
+    """Ladybug Date.
+
+    Properties:
+        month
+        day
+        leap_year
+        doy
+    """
+
+    __slots__ = ()
+
+    def __new__(cls, month=1, day=1, leap_year=False):
+        """Create Ladybug Date.
+
+        Args:
+            month: A value for month between 1-12 (Defualt: 1).
+            day: A value for day between 1-31 (Defualt: 1).
+            leap_year: A boolean to indicate if date is for a leap year
+                (Default: False).
+        """
+        year = 2016 if leap_year else 2017
+        try:
+            return date.__new__(cls, year, month, day)
+        except ValueError as e:
+            raise ValueError("{}:\n\t({}/{})(m/d)".format(e, month, day))
+
+    @classmethod
+    def from_dict(cls, data):
+        """Create date from a dictionary.
+
+        Args:
+            data: {
+                'month': A value for month between 1-12. (Defualt: 1)
+                'day': A value for day between 1-31. (Defualt: 1)
+            }
+        """
+        month = data['month'] if 'month' in data else 1
+        day = data['day'] if 'day' in data else 1
+        leap_year = data['leap_year'] if 'leap_year' in data else False
+        return cls(month, day, leap_year)
+
+    @classmethod
+    def from_doy(cls, doy, leap_year=False):
+        """Create Ladybug Date from an day of the year.
+
+        Args:
+            doy: An int value 0 <= and < 366
+        """
+        if not leap_year:
+            days_until_month = (0, 31, 59, 90, 120, 151, 181,
+                                212, 243, 273, 304, 334, 365)
+        else:
+            days_until_month = (0, 31, 60, 91, 121, 152, 182,
+                                213, 244, 274, 305, 335, 366)
+
+        for month_count in range(12):  # find month
+            if int(doy) < days_until_month[month_count + 1]:
```

Comment:
I would take `int(doy)` out of the loop. That will also ensure that your code in line 209 doesn't break. for illegal values you can use for / else pattern.

```python
doy = int(doy)
for month_count in range(12):  # find month
    if doy < days_until_month[month_count + 1]:
        month = month_count + 1
        break
else:
    raise ValueError(
        "doy must be positive and smaller than 366. Invalid input %d" % (doy)
    )

day = int(doy - days_until_month[month - 1])
```

Reply:
Good idea.  I will do the same thing for your old hoy code.